File: apps/java/src/main/java/com/functions/shared/AppConfig.java
package com.functions.shared;

/**
 * Centralized application configuration loaded from environment variables.
 */
public class AppConfig {

    private static final AppConfig INSTANCE = new AppConfig();

    private final String environment;
    private final String appName;
    private final String telemetryMode;

    private AppConfig() {
        String siteName = System.getenv("WEBSITE_SITE_NAME");
        this.appName = siteName != null ? siteName : "local";
        this.environment = siteName != null ? "production" : "development";
        String mode = System.getenv("TELEMETRY_MODE");
        this.telemetryMode = mode != null ? mode : "basic";
    }

    public static AppConfig getInstance() {
        return INSTANCE;
    }

    public String getEnvironment() {
        return environment;
    }

    public String getAppName() {
        return appName;
    }

    public String getTelemetryMode() {
        return telemetryMode;
    }

    public boolean isProduction() {
        return "production".equals(environment);
    }
}

File: apps/dotnet/Shared/AppConfig.cs
namespace AzureFunctionsGuide.Shared;

/// <summary>
/// Centralized application configuration from environment variables.
/// </summary>
public static class AppConfig
{
    public static string AppName => System.Environment.GetEnvironmentVariable("APP_NAME") ?? "azure-functions-dotnet-guide";
    public static string Version => "1.0.0";
    public static string FunctionAppName => System.Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME") ?? "local";
    public static string FunctionsEnvironment => System.Environment.GetEnvironmentVariable("AZURE_FUNCTIONS_ENVIRONMENT") ?? "production";
}
